docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+# ADR 111: `nil` Mempool
```

Comment:
Would it not be better to call this "Application-side mempool", given that that's the ultimate implication of this change?

Or is there some other way of emphasizing that that's the ultimate impact of this ADR?

Reply:
`abci` mempool ?

Reply:
The problem I have with "Application-side mempool", is that, for instance, what the SDK implemented in `v0.47.x` is also an "Application-side mempool", but (ATM) it does require a non-`nil` mempool on CometBFT's side.
I like like `nil`-mempool (or `null`-mempool, or stubbed mempool) as it points to what we are doing in CometBFT (not what the App can/must do when using our `nil`-mempool).